scripts/upgrade.sh

```diff
@@ -0,0 +1,203 @@
+#!/bin/bash
+
+#Author: Scality MetalK8s
+#
+#Summary: This script is used to upgrade Metalk8s cluster and all the components
+#          that make up the stack.
+#
+set -e
+set -u
+
+__ScriptName="$(basename "$0")"
+SALT_CALL=${SALT_CALL:-salt-call}
+
+VERBOSE=${VERBOSE:-0}
+LOGFILE=/var/log/metalk8s-upgrade.log
+
+# Some helpful guide on upgrade options
+__usage() {
+    cat << EOF
+  Usage :  ${__ScriptName} <options>
+  Options:
+    -h | --help              Display HELP options
+         --version           Show current installed version
+    -d | --dry-run           Show DRYRUN output.
+    -t | --to                Upgrade to specified version
+    -v | --verbose           Run in verbose mode
+EOF
+}
+
+
+TMPFILES=$(mktemp -d)
+
+cat << EOF >> "${LOGFILE}"
+--- MetalK8s Upgrade started on $(date -u -R) ---
+EOF
+
+exec > >(tee -ia "${LOGFILE}") 2>&1
+
+cleanup() {
+    rm -rf "${TMPFILES}" || true
+}
+
+trap cleanup EXIT
+
+
+run_quiet() {
+    local name=$1
+    shift 1
+
+    echo -n "> ${name}..."
+    local start
+    start=$(date +%s)
+    set +e
+    "$@" 2>&1 | tee -ia "${LOGFILE}" > "${TMPFILES}/out"
+    local RC=$?
+    set -e
+    local end
+    end=$(date +%s)
+
+    local duration=$(( end - start ))
+
+    if [ $RC -eq 0 ]; then
+        echo " done [${duration}s]"
+    else
+        echo " fail [${duration}s]"
+        cat >/dev/stderr << EOM
+
+Failure while running step '${name}'
+
+Command: $@
+
+Output:
+
+<< BEGIN >>
+EOM
+        cat "${TMPFILES}/out" > /dev/stderr
+
+        cat >/dev/stderr << EOM
+<< END >>
+
+This script will now exit
+
+EOM
+
+        exit 1
+    fi
+}
+
+run_verbose() {
+    local name=$1
+    shift 1
+
+    echo "> ${name}..."
+    "$@"
+}
+
+run() {
+    if [ "$VERBOSE" -eq 1 ]; then
+        run_verbose "${@}"
+    else
+        run_quiet "${@}"
+    fi
+}
+
+die() {
+    echo 1>&2 "$@"
+    return 1
+}
+
+__read_version() {
```

Comment:
I think this function (and the argument `--version`) is not relevant for the moment except if you show the version of all nodes because here you will only have the version of the bootstrap node

Reply:
ok